tests/ui_tools/test_boxes.py

```diff
@@ -1582,10 +1576,33 @@ def test_keypress_typeahead_mode_autocomplete_key(
         write_box.keypress(size, key)
 
         assert write_box.is_in_typeahead_mode == expected_typeahead_mode
-        if expect_footer_was_reset:
-            self.view.set_footer_text.assert_called_once_with()
-        else:
-            self.view.set_footer_text.assert_not_called()
+        self.view.set_footer_text.assert_not_called()
+
+    @pytest.mark.parametrize(
+        "key, current_typeahead_mode, expected_typeahead_mode",
+        [
+            (primary_key_for_command("GO_BACK"), True, False),
+            ("space", True, False),
+            ("k", True, False),
+        ],
+    )
+    def test_keypress_typeahead_mode_autocomplete_key_footer_reset(
```

Comment:
What's the simplification that can be done in terms of parametrize values? I'm a bit confused 😅.

Oh true, maybe we can should separate GO_BACK into its own test. Since reset actions are dependent on the length of the message in compose. What do you think?

Reply:
Re simplification: the keys vary, but the modes before and after apparently do not :)

Re my second comment, it likely needs covering, though may be worth being careful that it isn't over-tested, ie. in multiple places.

Reply:
The simplification is how the modes are the same in every case now, in each of these new tests.